nomenclate/core/nomenclature.py

```diff
@@ -2,146 +2,295 @@
+
+        if isinstance(input_object, self.nom.__class__):
+            input_object = input_object.state
+
+        input_object.update(kwargs)
+
+        for input_attr_name, input_attr_value in iteritems(input_object):
+            self.set_token_attr(input_attr_name, input_attr_value)
+
+    def build_name_attrs(self):
+        """ Creates all necessary name attributes for the naming convention
+        """
+        self.purge_invalid_name_attrs()
+
+        for token in self.nom.format_order:
+            self.set_token_attr(token, "")
+
+    def purge_invalid_name_attrs(self):
+        """ Removes name attrs not found in the format order
+        """
+        for token_attr in self.get_token_attrs():
+            try:
+                self._validate_name_in_format_order(token_attr.label, self.nom.format_order)
+            except exceptions.FormatError:
+                del self[token_attr.token]
+
+    def purge_name_attrs(self):
+        for token_attr in self.get_token_attrs():
+            del self[token_attr.token]
+        self.signal.fire()
+
+    def clear_name_attrs(self):
+        for token_attr in self.get_token_attrs():
+            token_attr.set('')
+
+    def set_token_attr(self, token, value):
+        token_attrs = self.get_token_attrs()
+        if token not in [token_attr.token for token_attr in token_attrs]:
+            self._create_token_attr(token, value)
+            self.signal.fire()
+        else:
+            for token_attr in token_attrs:
+                if token == token_attr.token:
+                    setattr(self, token_attr.token, value)
+
+    def get_token_attr(self, token):
+        token_attr = getattr(self, token.lower())
+        if token_attr is None:
+            raise exceptions.SourceError('This nomenclate instance has no %s attribute set.' % token)
+        else:
+            return token_attr
+
+    def get_token_attrs(self):
+        return [value for name, value in iteritems(self) if isinstance(value, TokenAttr)]
+
+    def _create_token_attr(self, token, value):
+        self[token.lower()] = TokenAttr(label=value, token=token)
+
+    @staticmethod
+    def _validate_name_in_format_order(name, format_order):
+        """ For quick checking if a key token is part of the format order
+        """
+        if name not in [token.lower() for token in format_order]:
+            raise exceptions.FormatError('The name token %s is not found in the current format ordering' %
+                                         format_order)
+
+    def __eq__(self, other):
+        print ('comparing token dicts self then other\n\t%s\n\t%s' % (self, other))
+        if isinstance(other, self.__class__):
+            equal = False
+            self_attrs = self.get_token_attrs()
+            other_attrs = other.get_token_attrs()
+            if len(self_attrs) > len(other_attrs):
+                return False
+
+            for attr in self_attrs:
```

Comment:
Surprise surprise, you did it the right way, now since they phased it out in py3.
http://stackoverflow.com/questions/27776026/lambda-arguments-unpack-error